apps/chat/src/components/Common/AgentAndToolsetSelector/SelectedItemsContainer.tsx

```diff
@@ -0,0 +1,43 @@
+import { MarketplaceEntity } from '@/src/types/marketplace';
+
+import { OverflowContainer } from '../OverflowContainer';
+import { AgentAndToolsetChip } from './AgentAndToolsetChip';
+import { OverflowButton } from './OverflowButton';
+
+interface SelectedItemsContainerProps {
+  selectedIds: string[];
+  allItemsMap: Record<string, MarketplaceEntity | undefined>;
+  onRemove: (id: string) => void;
+}
+
+const OVERFLOW_BUTTON_WIDTH = 50;
+
+export const SelectedItemsContainer = ({
+  selectedIds,
+  allItemsMap,
+  onRemove,
+}: SelectedItemsContainerProps) => {
+  const validItems = selectedIds
+    .map((id) => ({ id, data: allItemsMap[id] }))
+    .filter(
+      (item): item is { id: string; data: MarketplaceEntity } => !!item.data,
```

Comment:
do we need this typing conversion?
```suggestion
      (item): { id: string; data: MarketplaceEntity } => !!item.data,
```